openff/nagl_models/tests/test_dynamic_fetch.py

```diff
@@ -83,60 +45,94 @@ def hide_cache():
         shutil.move(alt_dir, cache_dir)
 
 
-def test_access_internet_with_empty_cache(hide_cache):
-    cache_path = platformdirs.user_cache_path() / "OPENFF_NAGL_MODELS"
+def test_zenodo_fetching_and_caching(hide_cache):
+    """
+    All of the tests that rely on remote fetching into the cache
+    and checking whether something is in the cache need to be run in
+    serial, otherwise they'll interfere with each other, so they're
+    all consolidated into this one test.
+    """
+
+    # This test uses a Zenodo sandbox DOI (10.5072 prefix) and the corresponding
+    # SHA256 hash of the test file "my_favorite_model.pt" (which is a copy of
+    # openff-gnn-am1bcc-0.1.0-rc.3.pt) uploaded to that sandbox record
+
+    from pytest_socket import SocketBlockedError, disable_socket, enable_socket
 
     disable_socket()
 
-    # would be nice to test the FileNotFoundError, but much more difficult to get that
-    # particular network failure vs. checking that the network is accessed at all
+    # Ensure that the cache is hidden,
+    with pytest.raises(FileNotFoundError):
+
+        get_model(
+            "my_favorite_model.pt",
+        )
+
+    # Ensure that trying to fetch a
+    # model fails due to lack of internet access
     with pytest.raises(
         SocketBlockedError,
     ):
-        get_model.cache_clear()
+        get_model(
+            "my_favorite_model.pt",
+            doi="10.5072/zenodo.278300",
+            file_hash="127eb0b9512f22546f8b455582bcd85b2521866d32b86d231fee26d4771b1d81",
+        )
 
-        get_model("openff-gnn-am1bcc-0.1.0-rc.3.pt")
+    # Ensure that the file can actually be fetched
+    enable_socket()
+    get_model(
+        "my_favorite_model.pt",
+        doi="10.5072/zenodo.278300",
+    )
 
+    # Ensure that, once fetched, the file can be gotten without accessing the internet.
+    disable_socket()
+    # Ensure that cached files can be accessed when no optional arguments are provided
+    get_model(
+        "my_favorite_model.pt",
+    )
```

Comment:
Could we be extra explicit about where the file is and isn't?

* Assert that this file is not in the installed Python package (I'd be surprised if it ever was) - feel free to do this anywhere in this test, I want to make sure that the checks into the cache and/or internet actually go there and aren't broken if we accidentally ship our "favorite" little model
* Assert that this file is present in the cache location

Reply:
Added!